lib/editor/components/ScheduleExceptionForm.js

```diff
@@ -233,7 +234,7 @@ export default class ScheduleExceptionForm extends Component<Props> {
               return (
                 <ExceptionDate
                   date={date}
-                  index={dates.findIndex(d => d === date)}
+                  index={dateIndex}
                   key={index}
```

Comment:
Maybe it makes sense, in this loop, to use uuids as keys, so that new date pickers are recreated when exceptions are combined (otherwise, if the index is the same but dates at that index change, components are not updated). Same thing when rendering the `ExceptionDateRange` elements above.

Reply:
Yes I used the uuids here, this works better than using them at the DateTimePicker level

Reply:
Updated as of https://github.com/ibi-group/datatools-ui/pull/843/commits/a83521ba6f5a2e66f8642193ca4a3c27efd71908